DashAISoftware/DashAI
# Experiment modal - last step

# Summary


Last step of the modal to create a new experiment, in this step models can be added and configured.


## Type of change



- Front end new feature.

## Changes


- The `ConfigureModelsStep.jsx` component is created as a container for the logic for the final step 
- A table (`ModelsTable.jsx`) is added to: 
  - display models that are currently in the experiment 
  - configure the parameters of a model (`EditModelDialog.jsx`) 
  - remove models from the experiment (`DeleteModelDialog.jsx`) 
- Some modifications are made to `ParameterForm.jsx` so that it can receive external initial values and thus be able to load the user's configurated values after having closed the configuration modal  
- The request to the experiment endpoint is updated 
- The `ClassInput.jsx` component is modified to support two options for recursive parameters: Modal or Collapsible

## How to Test


Option 1 - Backend
1. Go to `DashAI/DashAI/front` and run `yarn build`
2. Go back to `DashAI/` and run `python -c "import DashAI;DashAI.run()`
3. Navigate to the experiments tab
4.  Click the "New Experiment" button
5. Create a name and select a task for your experiment
6. Select the dataset for your experiment
7. Add a model to the table, optionally configure it and save the experiment.
8. Click "Update" and the new experiment should be visible in the table

## Screenshots


[createExperiment.webm](https://github.com/DashAISoftware/DashAI/assets/53657198/2e18d67d-4e7e-451a-b8b3-0c9fc9c56852)
